yb-voyager/cmd/importDataSequenceRestoration.go

```diff
@@ -0,0 +1,291 @@
+	if err != nil {
+		return fmt.Errorf("failed to restore sequences: %w", err)
+	}
+	return nil
+}
+
+func shouldFilterSequences(sourceType string) bool {
+	//if source type is not POSTGRESQL, we don't filter sequences
+	if sourceType != POSTGRESQL {
+		return false
+	}
+	//if table list filteration is enabled, we filter sequences
+	return tconf.TableList != "" || tconf.ExcludeTableList != ""
+}
+
+func fetchSequenceLastValueMap(isDebeziumExport bool, sourceType string) (map[string]int64, error) {
+	sequenceLastValue := make(map[string]int64)
+	var err error
+	if isDebeziumExport {
+		//read the sequence last value from the export status file if export is via debezium
+		status, err := dbzm.ReadExportStatus(filepath.Join(exportDir, "data", "export_status.json"))
+		if err != nil {
+			return nil, fmt.Errorf("failed to read export status for restore sequences: %w", err)
+		}
+		sequenceLastValue = status.Sequences
+	} else {
+		//read the sequence last value from the postdata.sql file if export is not via debezium
+		sequenceFilePath := filepath.Join(exportDir, "data", "postdata.sql")
+		if utils.FileOrFolderExists(sequenceFilePath) {
+			fmt.Printf("setting resume value for sequences %10s\n", "")
+			sequenceLastValue, err = readSequenceLastValueFromPostDataSql(sequenceFilePath, sourceType)
+			if err != nil {
+				return nil, fmt.Errorf("failed to read sequence last value from postdata.sql: %w", err)
+			}
+		}
+	}
+	return sequenceLastValue, nil
+}
+
+func fetchSequenceToTableListMap(sourceColumnToSequenceMapping map[string]string) (*utils.StructMap[sqlname.NameTuple, []string], error) {
+
+	sequenceNameToTableMap := utils.NewStructMap[sqlname.NameTuple, []string]()
+	for column, sequenceName := range sourceColumnToSequenceMapping {
+		parts := strings.Split(column, ".") //column is qualified tablename.colname
+		if len(parts) < 3 {
+			return nil, fmt.Errorf("invalid column name: %s", column)
+		}
+		tableName := fmt.Sprintf("%s.%s", parts[0], parts[1])
+		//get the sequence name from the source column to sequence mapping
+		//lookup the sequence name from source in name reg and ignore if target not found
+		sequenceTuple, err := namereg.NameReg.LookupTableNameAndIgnoreIfTargetNotFoundBasedOnRole(sequenceName)
+		if err != nil {
+			return nil, fmt.Errorf("error looking up sequence name %q: %w", sequenceName, err)
+		}
+		sequenceName = sequenceTuple.ForKey()
+		tableList, ok := sequenceNameToTableMap.Get(sequenceTuple)
+		if ok {
+			tableList = append(tableList, tableName)
+			sequenceNameToTableMap.Put(sequenceTuple, tableList)
+		} else {
+			sequenceNameToTableMap.Put(sequenceTuple, []string{tableName})
+		}
+	}
+	return sequenceNameToTableMap, nil
+}
+
+func checkIfSequenceAttachedToTablesInTableList(importTableList []sqlname.NameTuple, sequenceAttachedTables []string) (bool, error) {
+	for _, table := range sequenceAttachedTables {
+		tableTuple, err := namereg.NameReg.LookupTableNameAndIgnoreIfTargetNotFoundBasedOnRole(table)
```

Comment:
It feels like we're doing a name registry lookup in almost every function. That adds a bit of clutter everywhere. Can we do all the lookups in the beginning? 

For instance: 
1. fetchSequenceLastValueMap returns map[nametuple]int64
2. fetchSequenceToTableListMap returns map[nametuple] []nametuple
3. Then the filterSequencesAsPerImportTableList just has to do comparisons (no lookups)